dialect_mysql.go

```diff
@@ -72,8 +72,8 @@ func (d MySQLDialect) ToSqlType(val reflect.Type, maxsize int, isAutoIncr bool)
 		return "datetime"
 	}
 
-	if maxsize < 1 {
-		maxsize = 255
+	if maxsize < 1 || maxsize > 16384 {
```

Comment:
I don't love this change, we are introducing here an artificial constrain to the field size, when is not needed at all. We can use Text, MediumText and LongText here depending on the size of the field.

If you don't set an explicit limit, I would expect "unbounded" text, not arbitrary limited field. (because mysql doesn't have unbounded text field, I would use LongText). If you set a size limit that is over the VARCHAR limit, I would use a TEXT/MEDIUMTEXT/LONGTEXT field based on the size limit provided.

I agree that this is the only way to be fully consistentcy with postgres about size restrictions, but I think that is not needed, the underneath problem was initializing a "not size restriction field" in postgres and mysql and this discrepancy was huge, from 255 to unbounded.

Reply:
@lieut-data thoughs about this :point_up:

Reply:
I agree. Even this does not provide full consistency as mysql will silently truncate a string exceeding the column max (unless server is set to strict SQL mode) while postgres will trigger an error.

Here's an alternative that uses _longtext_ for maxsize=0, _varchar_ for maxsize <= 512 and _text_ if maxsize > 512.
No change to the postgres dialect.
Of course it's still not perfect as mysql won't enforce the specified size constraint if it's higher than 512. Still, I think it's a good compromise.

Let me know what you think.

https://github.com/mattermost/gorp/pull/5

Reply:
Ah, It looks like I reviewed https://github.com/mattermost/gorp/pull/5 before this one, which explains my confusion on the original plan posted to the ticket!

Reply:
I think the migration issue is key to whatever decision we make here. I'm still worried about (eventually) forcing a column change to existing databases. It may prove to be that the answer to the original ticket is "no, don't change" because there's no good outcome.

@jespino, @brunoenten -- let's continue the conversation in whatever PR seems most suitable :)

Reply:
Ok, from my perspectiv, this approach have this "migrations":
  - for the users of mysql will change:
    - any "not max size defined" field from `VARCHAR(255)` to `VARCHAR(16384)`,
    - any field with the max size over `16384` to a from a `TEXT` to `VARCHAR(16384)`
    - and field between 256 and 16384 from a `TEXT` to `VARCHAR(X)`
  - for the users of postgres will change:
    - any "not max size defined" field from `TEXT` to `VARCHAR(16384)`,
    - any field with the max size over `16384` to a from a `VARCHAR(X)` to `VARCHAR(16384)`

With the #5 approach:
  - for the users of mysql will change:
    - any "not max size defined" field from `VARCHAR(255)` to `LONGTEXT`,
    - any field with the max size between 256 and 512 to a from a `TEXT` to `VARCHAR(X)` (probably we should keep the boundary in 256 as was before to avoid this extra migration.
  - for the users of postgres no changes needed.

With this information, I think we are in a better position with the PR #5, so, lets keep talking there.